### PR TITLE
Null Ranking comparator error handling

### PR DESCRIPTION
**Pull request by mweiss17**
#### What’s this PR do?
- The comparator method now tolerates null rankings gracefully. It puts contacts with null rankings at the bottom of the list.

#### Where should the reviewer start?
Run the tests

#### How should this be manually tested?
debug/step through the compare(RankedContact p1, RankedContact p2) method in AddressBook.java and isAscOrdered(ContactList contactsList) in AddressBookUnitTest.java

#### Any background context you want to provide?
Ranked contact objects should not have a null ranking.

### CODE REVIEW

File: yesgraph/src/androidTest/java/com/yesgraph/android/AddressBookUnitTest.java [hunk 1]
```diff
@@ -198,7 +238,13 @@ private boolean isAscOrdered(ContactList contactsList) {
 
             RankedContact contact = contactsList.getEntries().get(i);
             RankedContact nextContact = contactsList.getEntries().get(i + 1);
-
+            if (contact.getRank() == null && !(nextContact.getRank() == null)){
+                isOrdered = false;
+                break;
+            }
+            if (contact.getRank() == null || nextContact.getRank() == null){
```
Comment (kendallc) [inline yesgraph/src/androidTest/java/com/yesgraph/android/AddressBookUnitTest.java]: Would use an "else" here

File: yesgraph/src/androidTest/java/com/yesgraph/android/AddressBookUnitTest.java [hunk 2]
```diff
@@ -198,7 +238,13 @@ private boolean isAscOrdered(ContactList contactsList) {
...
+                break;
+            }
             if (contact.getRank() > nextContact.getRank()) {
```
Comment (kendallc) [inline yesgraph/src/androidTest/java/com/yesgraph/android/AddressBookUnitTest.java]: Also an "else if" here

File: yesgraph/src/androidTest/java/com/yesgraph/android/AddressBookUnitTest.java [hunk 3]
```diff
@@ -198,7 +238,13 @@ private boolean isAscOrdered(ContactList contactsList) {
 
             RankedContact contact = contactsList.getEntries().get(i);
             RankedContact nextContact = contactsList.getEntries().get(i + 1);
-
+            if (contact.getRank() == null && !(nextContact.getRank() == null)){
+                isOrdered = false;
+                break;
+            }
+            if (contact.getRank() == null || nextContact.getRank() == null){
+                break;
```
Comment (kendallc) [inline yesgraph/src/androidTest/java/com/yesgraph/android/AddressBookUnitTest.java]: Also probably continue instead of break here. What happens if we have "1, 2, 3, null, null, 4, 5, 6"?

File: yesgraph/src/androidTest/java/com/yesgraph/android/AddressBookUnitTest.java [hunk 4]
```diff
@@ -91,6 +91,46 @@ public void testValidateContactsListOrderedAsc() throws JSONException {
...
+        // Set  both to null
+        jsonContacts = new JSONArray();
+        jsonContact0.put("rank", null);
+        jsonContact1.put("rank", null);
+        jsonContacts.put(jsonContact0);
+        jsonContacts.put(jsonContact1);
+        ysgContactList = new AddressBook().contactListFromResponse(jsonContacts);
+        areAscOrdered = isAscOrdered(ysgContactList);
+        assertEquals(true, areAscOrdered);
+
```
Comment (kendallc) [inline yesgraph/src/androidTest/java/com/yesgraph/android/AddressBookUnitTest.java]: We should probably also test the case where you have null contacts in between ranked contacts. "1, 2, null, null, 3, 4"

Comment (kendallc) [inline yesgraph/src/androidTest/java/com/yesgraph/android/AddressBookUnitTest.java]: Also can we test a failing case before ranking the contacts? e.g. "4, 3, null, 1" is not ranked (areAscOrdered is false), then "1, 3, 4, null" is ranked
